piff/psf.py

```diff
@@ -722,34 +719,34 @@ def write(self, file_name, logger=None):
         :param file_name:   The name of the file to write to.
         :param logger:      A logger object for logging debug info. [default: None]
         """
+        from .writers import Writer
+
         logger = galsim.config.LoggerWrapper(logger)
         logger.warning("Writing PSF to file %s",file_name)
 
-        with fitsio.FITS(file_name,'rw',clobber=True) as f:
-            self._write(f, 'psf', logger)
+        with Writer.open(file_name) as w:
+            self._write(w, 'psf', logger=logger)
 
-    def _write(self, fits, extname, logger=None):
+    def _write(self, writer, name, logger):
```

Comment:
Throughout this ticket I've taken the liberty of just changing the signatures of underscore-prefixed methods (`_write`, `_finish_write`, `_read`, `_finish_read`) but I've kept `read` and `write` the same (while having them delegate to the new methods.  If any of those underscore-prefixed methods shouldn't change because they might be in use outside Piff, I can go back and add new methods instead.